From 7f9c8714eb4ca307441a0997ebaef65514003df0 Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Sat, 4 May 2024 08:37:40 +0530
Subject: [PATCH] Mark USB camera to back camera always

Google camera preview show mirror image when switching between
two USB camera.

In case of 2 camera use-case app considering front and back camera
and based on this app is mirroring front camera preview. exposing
both camera as back camera to framework so mirrors wont happen.

Tracked-On: OAM-117469
Signed-off-by: shivasku82 <shiva.kumara.rudrappa@intel.com>
---
 camera/device/3.4/default/ExternalCameraDevice.cpp | 2 +-
 camera/device/default/ExternalCameraDevice.cpp     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/camera/device/3.4/default/ExternalCameraDevice.cpp b/camera/device/3.4/default/ExternalCameraDevice.cpp
index 44b908b289..1a6ee2a8b8 100644
--- a/camera/device/3.4/default/ExternalCameraDevice.cpp
+++ b/camera/device/3.4/default/ExternalCameraDevice.cpp
@@ -370,7 +370,7 @@ status_t ExternalCameraDevice::initDefaultCharsKeys(
     UPDATE(ANDROID_LENS_INFO_AVAILABLE_OPTICAL_STABILIZATION,
            &opticalStabilizationMode, 1);
 
-    const uint8_t facing = ANDROID_LENS_FACING_EXTERNAL;
+    const uint8_t facing = ANDROID_LENS_FACING_BACK;
     UPDATE(ANDROID_LENS_FACING, &facing, 1);
 
     // android.noiseReduction
diff --git a/camera/device/default/ExternalCameraDevice.cpp b/camera/device/default/ExternalCameraDevice.cpp
index 8e8474800f..50fbdd4efe 100644
--- a/camera/device/default/ExternalCameraDevice.cpp
+++ b/camera/device/default/ExternalCameraDevice.cpp
@@ -420,7 +420,7 @@ status_t ExternalCameraDevice::initDefaultCharsKeys(
     const uint8_t opticalStabilizationMode = ANDROID_LENS_OPTICAL_STABILIZATION_MODE_OFF;
     UPDATE(ANDROID_LENS_INFO_AVAILABLE_OPTICAL_STABILIZATION, &opticalStabilizationMode, 1);
 
-    const uint8_t facing = ANDROID_LENS_FACING_EXTERNAL;
+    const uint8_t facing = ANDROID_LENS_FACING_BACK;
     UPDATE(ANDROID_LENS_FACING, &facing, 1);
 
     // android.noiseReduction
-- 
2.45.2

